### ISSUE TITLE
Tracking issue for filter expressions

### ISSUE THREAD

**Issue by sunshowers** [nextest-rs/nextest#158]
Documentation: https://nexte.st/book/filter-expressions.html

TODO:
* [x] #159
* [x] #160
* [x] #29

cc @Guiguiprim

**Comment by pan93412** [comment]
I have such a test:

```
crypto-markets::bitmex aaaaa
crypto-markets::bitmex bbbbb
crypto-markets::bitmex ccccc
crypto-markets::others bbbbb
crypto-markets::dont_exclude aaaaa
```

How can I exclude the `bitmex` tests while still including `others` and `dont_exclude`? I tried `package(/bitmex/)`, and it said:

```
$ cargo nextest run -E 'package(/bitmex/)'

  error: operator didn't match any packages
   ╭────
 1 │ package(/bitmex/)
   ·         ────┬───
   ·             ╰── no packages matched this
   ╰────

error: failed to parse filter expression
```

While `test(/bitmex/)` can only match:

```
crypto-markets::bitmex bitmex_aaaaa
crypto-markets::bitmex bitmex_bbbbb
crypto-markets::bitmex bitmex_ccccc
crypto-markets::others bitmex_bbbbb
crypto-markets::dont_exclude bitmex_aaaaa
```

**Comment by sunshowers** [comment]
Thanks for pointing this out! The next version of nextest (hopefully done in a couple of days) will add a new `binary` predicate for this purpose.

**Comment by sunshowers** [comment]
Filter expressions have been released and are now part of cargo-nextest 0.9.25.
